packages/js/client/src/createPolywrapClient.ts

```diff
@@ -71,10 +72,17 @@ export const createPolywrapClient = Tracer.traceFunc(
     }
 
     if (config) {
-      return new PolywrapClient({
-        ...config,
-        plugins: [...plugins, ...(config.plugins ? config.plugins : [])],
-      });
+      const builder = new ClientConfigBuilder().add(config);
+
+      if (plugins.length) {
```

Comment:
Do we need to do this validation? :thinking:

Reply:
Nah, not really, you're right.

This is also a helper function which I'd honestly like to get rid of 😄
While it seems useful, the config builder kind of defeats its purpose. Tests can be refactored to not use this if we choose to do so.